src/pint/pint_matrix.py

```diff
@@ -50,6 +51,12 @@ def __init__(self, matrix, axis_labels):
         # Check label index overlap TODO: should we allow overlap?
         self._check_index_overlap()
 
+    def __getitem__(self, key):
```

Comment:
Could you add a docstring here to explain what `key` is supposed to be? Like, is this supposed to implement array indexing, or indexing by label, or both?

Reply:
Done.  Also removed `__setitem__` since I realized it wouldn't make sense and could lead to weird behavior.